lib/iris/fileformats/pp.py

```diff
@@ -1378,8 +1378,18 @@ def save(self, file_handle):
 
         # Get the actual data content.
         data = self.data
-        if ma.is_masked(data):
-            data = data.filled(fill_value=self.bmdi)
+        mdi = self.bmdi
+        if np.any(data == mdi):
+            msg = ('PPField data contains unmasked points equal to the fill '
+                   "value, {}. As saved, these points will read back as "
+                   "missing data. To save these as normal values, please "
+                   "set the field BMDI not equal to any valid data points.")
```

Comment:
Are you not allowing the BMDI to be specified via a keyword to `iris.save`?

Reply:
On consideration, I didn't think it was necessary.
It's not the same as the netdcf case, because we don't have to worry about smaller integer types.  So, especially if we *don't* provide an option to easily change BMDI, then chance collisions are extremely unlikely.  Likewise, more recent version of the UM spec ("F3 document") don't actually admit the use of non-standard BMDI values.

Reply:
I'm still not completely sold - I think there's an argument for providing an a uniform interface for both PP and NetCDF. But I think it could easily be changed later (post Iris 2 release) without breaking anything so I won't worry about it now.